app/models/permit_application.rb

```diff
@@ -29,8 +46,48 @@ def requirements
     requirement_template ? requirement_template.to_form_json : nil
   end
 
+  def number_prefix
+    jurisdiction.prefix
+  end
+
+  def assign_unique_number
+    last_number =
+      jurisdiction
+        .permit_applications
+        .where("number LIKE ?", "#{number_prefix}-%")
+        .order(Arel.sql("LENGTH(number) DESC"), number: :desc)
+        .limit(1)
+        .pluck(:number)
+        .first
+
+    if last_number
+      number_parts = last_number.split("-")
+      sequence_part = number_parts[1..-1].join.to_i + 1 # Increment the sequence
+      new_number =
+        format(
+          "%s-%03d-%03d-%03d",
+          number_prefix,
+          sequence_part / 1_000_000 % 1000,
+          sequence_part / 1000 % 1000,
+          sequence_part % 1000,
+        )
+    else
+      # Start with the initial number if there are no previous numbers
+      new_number = format("%s-001-000-000", number_prefix)
+    end
+
+    # Assign the new number to the permit application
+    self.number = new_number
+    return new_number
+  end
```

Comment:
What is the goal of this? Can you give some examples

Reply:
Sure!:

The objective is to create a unique number for each application.

like CN-001-111-111
then CN-001-111-112

I may need to test this more but initial results are good

Reply:
im going to fix up the clarity of this a bit

Reply:
![image](https://github.com/bcgov/HOUS-permit-portal/assets/19396915/e3ac8f7e-6617-4b79-881e-e4471d0a3597)